onc-healthit/lantern-back-end
# Add user agent to request header

VERSION file is mounted to capability querier container and version is read within 
Format for user-agent should be application/version
Lantern user-agent is now LANTERN/0.9.0 

To test:
Start with empty db
`make run`

Go to https://webhook.site/
and copy the unique url. This will be the test endpoint which we can view our http request

In pgAdmin 4 insert a test endpoint
```
insert into fhir_endpoints(url, organization_names, npi_ids, list_source) values
('http://your-copied-url.com', '{}', '{}', 'test');
```
run the sendendpoints and go back to https://webhook.site/
In the left column you will see incoming request. There should be several request (from metadata, well-known, and different mime types). Click on them and check that the user-agent in the header is `Lantern/0.9.0`

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is:  https://oncprojectracking.healthit.gov/support/browse/LANTERN-226
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
